neqo-transport/src/frame.rs

```diff
@@ -426,8 +426,13 @@ impl<'a> Frame<'a> {
             d(dec.decode_varint())
         }
 
-        // TODO(ekr@rtfm.com): check for minimal encoding
+        // Check for minimal encoding of frame type.
+        let pos = dec.offset();
         let t = dv(dec)?;
+        if Decoder::minimal_varint_len(t).ok_or(Error::ProtocolViolation)? != dec.offset() - pos {
```

Comment:
```suggestion
        if Decoder::minimal_varint_len(t).unwrap() != dec.offset() - pos {
```
should also be safe, given that we just decoded `t`.